ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateProcessor.java

```diff
@@ -86,13 +89,20 @@ public Bytes32 processHead() {
   @Subscribe
   public void onDeposit(tech.pegasys.artemis.pow.event.Deposit event) {
     STDOUT.log(Level.DEBUG, "New deposit received");
-    deposits.add(DepositUtil.convertDepositEventToOperationDeposit(event));
+    final DepositWithIndex depositWithIndex =
+        DepositUtil.convertDepositEventToOperationDeposit(event);
+    deposits.add(depositWithIndex);
+    // Eth1 hash has to be from the block containing the last required deposit but we may be
+    // receiving deposits out of order.
+    if (deposits.last().equals(depositWithIndex)) {
+      genesisEth1BlockHash = Bytes32.fromHexString(event.getResponse().log.getBlockHash());
```

Comment:
It's probably clearer to just work with the deposit object:
```suggestion
      genesisEth1BlockHash = Bytes32.fromHexString(depositWithIndex.getLog().getBlockHash());
```